spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -202,8 +202,39 @@
 
     end
 
-  end
+    context "429 errors" do
+      let(:event) { ::LogStash::Event.new("foo" => "bar") }
+      let(:error) do 
+        ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError.new(
+          429, double("url").as_null_object, double("request body"), double("response body")
+        )
+      end
+      let(:logger) { double("logger").as_null_object }
+
+      before(:each) do
+        i = 0
+        bulk_param =  [["index", anything, event.to_hash]]
+
+        allow(eso).to receive(:logger).and_return(logger)
+        allow(eso.client).to receive(:bulk).with(bulk_param) do
```

Comment:
Ahh, thank you for that clarity. I understand the intent now. Can you add a coment above the line to explain this intent for future readers?